packages/app/src/D2Shim.js

```diff
@@ -11,26 +11,28 @@ import { apiFetchOrganisationUnitLevels } from './api/organisationUnits'
 import history from './modules/history'
 import { ScreenCover, CircularLoader } from '@dhis2/ui-core'
 
-const configI18n = async () => {
+const configI18n = async userSettings => {
+    const uiLocale = userSettings.uiLocale
+
     // TODO: Remove and port to modern i18n
-    // if (uiLocale && uiLocale !== 'en') {
-    //     config.i18n.sources.add(
-    //         `./i18n_old/i18n_module_${uiLocale}.properties`
-    //     );
-    // }
+    if (uiLocale && uiLocale !== 'en') {
+        d2Config.i18n.sources.add(
+            `./i18n_old/i18n_module_${uiLocale}.properties`
+        )
+    }
 
     d2Config.i18n.sources.add('./i18n_old/i18n_module_en.properties')
 }
 
 const initD2 = async ({ baseUrl, apiVersion }, initConfig) => {
+    const userSettings = extractUserSettings(await getUserSettings())
```

Comment:
This won't work because `d2` hasn't been initialized with the correct baseUrl (it gets initialized 2 lines below)

Reply:
Or at least this *shouldn't* work.... were you testing on localhost in development?

Reply:
Yes, I tested on localhost.
I think it can also be that it works for english as that's the fallback anyway.

Tested now after switching to a different language and saw the invalid request to `userSettings` which was not setting the locale, thus no propeties file was loaded for the user locale.

I looked again at the code in v33 and added a commit with a fix.
Now it basically does the same as v33, where `d2.config.baseUrl` is set before calling `userSettings`.

Reply:
Nice, yeah I think that should work!